src/meili/routing.cc

```diff
@@ -562,6 +567,16 @@ find_shortest_path(baldr::GraphReader& reader,
                                 other_edge.percent_along, cost, turn_cost, cost.cost, label_idx,
                                 directededge, travelmode);
                 }
+              } else if (opp_edgeid == other_edge.id &&
```

Comment:
Some readability feedback: we're using a condition in this `if` branch above this and this `else if` branch. Can we refactor this into a bool variable with a meaningful name?  This would make the code more self-explanatory. I'm catching up on what file is doing, so I'm making up a name here. Choose something you think is appropriate.

```c++
const bool has_forward_match = origin_edge.id == other_edge.id && origin_edge.percent_along <= other_edge.percent_along
const bool has_reverse_edge_match = opp_edgeid == other_edge.id && (1.0f - origin_edge.percent_along) <= other_edge.percent_along;

if (has_forward_match) {
  // ..
} else if (has_reverse_edge_match) {
  // ..
} else {
  // Should we handle this case too? 
}
```

Reply:
> ```c++
> const bool has_forward_match = origin_edge.id == other_edge.id && origin_edge.percent_along <= other_edge.percent_along
> const bool has_reverse_edge_match = opp_edgeid == other_edge.id && (1.0f - origin_edge.percent_along) <= other_edge.percent_along;
> 
> if (has_forward_match) {
>   // ..
> } else if (has_reverse_edge_match) {
>   // ..
> } else {
>   // Should we handle this case too? .. or indicate something else, or return an error, etc.
> }
> ```
I think for brevity, we can skip using `const` for short-lived local variables. Also I would prefer putting parenthesis around the long condition for evaluating the bool to make it more readable. Please ignore this comment, if we are not implementing the suggestion that @mookerji pointed out.